pkg/types/rfc3161/rfc3161.proto

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v2.types.rfc3161;
+
+import "google/api/field_behavior.proto";
+
+option go_package = "github.com/sigstore/rekor/pkg/generated/protobuf/types/rfc3161";
+option java_package = "dev.sigstore.rekor.v2.types.rfc3161";
+option java_multiple_files = true;
+option java_outer_classname = "RekorRFC3161Proto";
+
+/*
+ * ProposedTimestampRequest represents the structure of the message required to present to Rekor
+ * to make an entry of the TimestampRequest type.
+ *
+ * TimestampRequest uses an identical input and output format;
+ * we create both for potential forward compatibility of separating and changing them;
+ * this also makes the assignment more explicit
```

Comment:
I'm a little confused by this - So ProposedTImestampRequest is the request, and TimestampRequest is an internal type?

Reply:
no, for all types I want to explicitly clarify that they must send `Proposed_______` and they will get `________` as a response. One of the major downsides to the current JSON-schema approach is that it is difficult to interpret what will actually get persisted in the log (by looking at `readOnly` or `writeOnly` tags in JSON schema)... Having two separate types allows for explicit hydration in the code.

Reply:
this is a typo, good catch